backstory/src/demo/java/com/google/sps/servlets/AnalyzeImageServlet.java

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Text;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.protobuf.ByteString;
+import com.google.sps.images.data.AnnotatedImage;
+import com.google.sps.images.ImagesManager;
+import com.google.sps.images.VisionImagesManager;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Prototype Servlet for serving the POST request for the analyze image resource. It uses the Vision
+ * API to analyze the images.
+ */
+@WebServlet("/analyze-image")
+public class AnalyzeImageServlet extends HttpServlet {
+  /**
+   * {@inheritDoc}
+   *
+   * Expecting a post request from Blobstore containing the data fields from the image-upload
+   * form. After having gone through Blobstore, the request will include the image uploaded. The
+   * form in the HTML will connect to the Blobstore URL, which encodes the image and then
+   * redirects the request to this Url. In this doPost, the image is analyzed with Vision API,
+   * and the analysis along with the image URL will be sent to datastore for permanent storage.
+   */
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
```

Comment:
What does get the input from the form refer to? I can't seem to locate the code it's describing.